lightning/src/chain/keysinterface.rs

```diff
@@ -51,43 +51,61 @@ pub enum SpendableOutputDescriptor {
 		output: TxOut,
 	},
 	/// An output to a P2WSH script which can be spent with a single signature after a CSV delay.
-	/// The private key which should be used to sign the transaction is provided, as well as the
-	/// full witness redeemScript which is hashed in the output script_pubkey.
+	///
 	/// The witness in the spending input should be:
-	/// <BIP 143 signature generated with the given key> <empty vector> (MINIMALIF standard rule)
-	/// <witness_script as provided>
-	/// Note that the nSequence field in the input must be set to_self_delay (which corresponds to
-	/// the transaction not being broadcastable until at least to_self_delay blocks after the input
-	/// confirms).
+	/// <BIP 143 signature> <empty vector> (MINIMALIF standard rule) <provided witnessScript>
+	///
+	/// Note that the nSequence field in the spending input must be set to to_self_delay
+	/// (which means the transaction not being broadcastable until at least to_self_delay
+	/// blocks after the outpoint confirms).
+	///
 	/// These are generally the result of a "revocable" output to us, spendable only by us unless
 	/// it is an output from us having broadcast an old state (which should never happen).
+	///
+	/// WitnessScript may be regenerated by passing the revocation_pubkey, to_self_delay and
+	/// delayed_payment_pubkey to chan_utils::get_revokeable_redeemscript.
+	///
+	/// To derive the delayed_payment key corresponding to the channel state, you must pass the
+	/// local delayed_payment_base_key and the provided per_commitment_point to
+	/// chan_utils::derive_private_key. The resulting key should be used to sign the spending
+	/// transaction.
+	///
+	/// To derive the revocation_pubkey corresponding to the channel state, you must pass the
+	/// remote revocation_basepoint and the provided per_commitment point to
+	/// chan_utils::derive_public_revocation_key.
+	///
+	/// Both remote revocation_basepoint and local delayed_payment_base_key should be given
+	/// by ChannelKeys, either default implementation (InMemoryChannelKeys) or custom one.
 	DynamicOutputP2WSH {
 		/// The outpoint which is spendable
 		outpoint: OutPoint,
-		/// The secret key which must be used to sign the spending transaction
-		key: SecretKey,
-		/// The witness redeemScript which is hashed to create the script_pubkey in the given output
-		witness_script: Script,
+		/// Per commitment point to derive delayed_payment_key by key holder
+		per_commitment_point: PublicKey,
 		/// The nSequence value which must be set in the spending input to satisfy the OP_CSV in
 		/// the witness_script.
 		to_self_delay: u16,
 		/// The output which is referenced by the given outpoint
 		output: TxOut,
+		/// The channel keys state used to proceed to derivation of signing key. Must
+		/// be pass to KeysInterface::derive_channel_keys.
+		key_state: (u64, u64),
```

Comment:
Key "state" is a strange name IMO. Its not really "state" associated with a key, more like key_derivation_parameters or channel_keys_id or so. Also please use consistent naming everywhere - pick one and don't reference chan_id/stating_time_* etc, just call it one thing.